source/MaterialXCore/Document.h

```diff
@@ -339,9 +339,9 @@ class MX_CORE_API Document : public GraphElement
     /// @param node Node type for the new declaration
     /// @param version Version for the new declaration
     /// @param isDefaultVersion If a version is specified is thie definition the default version
+    /// @param nodeGroup Optional node group for the new declaration. The Default value is an emptry string.
     /// @param newGraphName Make a copy of this NodeGraph with the given name if a non-empty name is provided. Otherwise
     ///        modify the existing NodeGraph. Default value is an empty string.
```

Comment:
@kwokcb Maybe it's just me, but I'm still struggling with the idea that `addNodeDefFromGraph` modifies the graph itself, which seems very unexpected from a user perspective.  Could these two steps be decoupled, so that `addNodeDefFromGraph` treats the input graph as constant, with a separate non-const method that converts a compound graph into a functional graph?

As far as I know, there aren't yet any community examples of tools that leverage the original implementation of `addNodeDefFromGraph`, so I think we're in a good position to refactor and simplify it, and we can confirm on the MaterialX Slack before proceeding.

Reply:
Hmm. This feels wrong to me -- and I don't recall why this logic was added...
So as you say this method will always add in a new nodegraph. It will still fail if a graph of the same name exists and no unique name creation will occur (as is the case now).

Having a second version which converts an existing compound to a functional can be added later if needed as you suggest.

Reply:
I've simplified the interface as much as I can see possible. The minimal is:
1. A compound graph
2. A new nodedef name and category
3. A new functional graph name
I removed version and group as these are not necessary.